Add `dangerouslySetTSConfig` sku config option (#848)
seek-oss/sku: .changeset/tsconfig-dangerously.md

---
'sku': minor
---

> `srcPaths` no longer affects `tsconfig.json#include`. Instead, you can use [`dangerouslySetTSConfig`][dangerous] option to have more control over which files are included in the TypeScript compilation process.

Reviewer: ```suggestion
`srcPaths` no longer affects `tsconfig.json#include`. Instead, you can use the [`dangerouslySetTSConfig`][dangerous] option to have more control over which files are included in the TypeScript compilation process.
```

> `srcPaths` no longer affects `tsconfig.json#include`. Instead, you can use the [`dangerouslySetTSConfig`][dangerous] option to have more control over which files are included in the TypeScript compilation process.

Reviewer: Just worth clarifying this, since we don't actually use `tsc` for compilation.
```suggestion
`srcPaths` no longer affects `tsconfig.json#include`. Instead, you can use the [`dangerouslySetTSConfig`][dangerous] option to have more control over which files are included in the typechecking process.
```

> Previously, sku managed the `include` field in `tsconfig.json`, but this was problematic for projects that wanted more fine grained control over what is included and/or excluded from compilation.

Reviewer: I _think_ this should be `was` so it matches the tense of the rest of the sentence, e.g. `wanted`.
```suggestion
Previously, sku managed the `include` field in `tsconfig.json`, but this was problematic for projects that wanted more fine grained control over what was included and/or excluded from compilation.
```

> > **Note**: If you were previously using [`srcPaths`][srcpaths] for this purpose, you should remove the paths which are not source files.

Reviewer: Maybe this should be turned into more of a migration guide rather than just a note. Though I guess it might be hard to provide a generic migration guide as it won't necessarily be the same for all consumers. 🤔
